apps/checker/app/services/SentenceHelpers.scala

```diff
@@ -43,8 +43,13 @@ class SentenceHelpers() {
   def maybeGetFirstWordFromSentence(sentence: CoreMap) = {
     val tokens = sentence.get(classOf[TokensAnnotation]).asScala.toList
     val maybeFirstValidToken = tokens.find { token =>
-      !SentenceHelpers.NON_WORD_TOKENS.contains(token.value()) &&
-      !SentenceHelpers.NON_WORD_TOKEN_CHARS.exists(token.value().contains(_))
+      val tokenDoesNotContainNonWordToken = !SentenceHelpers.NON_WORD_TOKENS.contains(token.value())
+      val tokenWithNonWordCharsRemoved = SentenceHelpers.NON_WORD_TOKEN_CHARS.foldLeft(token.value()) {
+        (remainingToken, nonWordChar) => remainingToken.replace(nonWordChar, "")
```

Comment:
I feel like this logic would be cleaner and clearer with some well named functions such as `stripNonWordCharPrefix` doing the heavy lifting. Also I think something like `token.value().filter` might be clearer as a `String` can be treated as a sequence of `Char` in Scala (easier if your `NON_WORD_TOKEN_CHARS` is a Set[Char]).

```suggestion
      val tokenWithNonWordCharsRemoved = token.value().filterNot(SentenceHelpers.NON_WORD_TOKEN_CHARS.map(_.head).toSet.contains)
```

Reply:
Agreed – I've made amends per your suggestion. Good shout on `filter`. No reason for both lists of chars not to be Sets, which presumably makes `contains` quicker.